src/Catalyst.Core.Lib/P2P/Repository/IPeerRepository.cs

```diff
@@ -18,16 +17,36 @@
 * You should have received a copy of the GNU General Public License
 * along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
 */
-
 #endregion
 
-using Catalyst.Abstractions.Repository;
-using Catalyst.Core.Lib.DAO.Peer;
 using Catalyst.Core.Lib.P2P.Models;
+using Google.Protobuf;
+using System.Collections.Generic;
 
 namespace Catalyst.Core.Lib.P2P.Repository
 {
-    public interface IPeerRepository : IRepositoryWrapper<Peer> { }
+    public interface IPeerRepository
```

Comment:
You can do it this way but then you'll need to modify this (because it is populating them inside the container using reflection and not checking if base interface is IRepository):
https://github.com/SharpRepository/SharpRepository/blob/34c910bdf480d40cf5f8ea525960d4caeaca54f6/SharpRepository.Ioc.Autofac/RepositoryRegistrationSource.cs

```suggestion
    public interface IPeerRepository : IRepository<Peer, string>
```
Only benefit from this is that you wont get IRepository<Peer, string> all over the code.

Reply:
Yeah, I have talked with James a few days ago about it and how the Wrapper works by allowing you to use the full repository functionality and think we are going to only allow users to construct their own specific methods and limit direct querying.